src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,186 @@
+
+		var inputs = userGeneratedClassesProvider.Collect()
+						.Combine(mauiControlsAssemblySymbolProvider)
+						.Select(static (combined, cancellationToken) => (UserGeneratedClassesProvider: combined.Left, MauiControlsAssemblySymbolProvider: combined.Right))
+						.Combine(context.CompilationProvider)
+						.Select(static (combined, cancellationToken) => (combined.Left.UserGeneratedClassesProvider, combined.Left.MauiControlsAssemblySymbolProvider, Compilation: combined.Right));
+
+		context.RegisterSourceOutput(inputs, (context, collectedValues) =>
+		Execute(context, collectedValues.Compilation, collectedValues.UserGeneratedClassesProvider, collectedValues.MauiControlsAssemblySymbolProvider));
+	}
+
+	static void Execute(SourceProductionContext context, Compilation compilation, ImmutableArray<ClassDeclarationSyntax> userGeneratedClassesProvider, IAssemblySymbol mauiControlsAssemblySymbolProvider)
+	{
+		var textStyleSymbol = compilation.GetTypeByMetadataName(iTextStyleInterface);
+		var iAnimatableSymbol = compilation.GetTypeByMetadataName(iAnimatableInterface);
+
+		if (textStyleSymbol is null || iAnimatableSymbol is null)
+		{
+			var diag = Diagnostic.Create(TextColorToDiagnostic.MauiReferenceIsMissing, Location.None);
+			context.ReportDiagnostic(diag);
+			return;
+		}
+
+		var textStyleClassList = new List<(string ClassName, string ClassAcessModifier, string Namespace)>();
+
+		// Collect Microsoft.Maui.Controls that Implement ITextStyle
+		var mauiTextStyleImplementors = mauiControlsAssemblySymbolProvider.GlobalNamespace.GetNamedTypeSymbols().Where(x => x.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+				&& x.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default));
+
+		foreach (var namedTypeSymbol in mauiTextStyleImplementors)
+		{
+			textStyleClassList.Add((namedTypeSymbol.Name, "public", namedTypeSymbol.ContainingNamespace.ToDisplayString()));
+		}
+
+		// Collect All Classes in User Library that Implement ITextStyle
+		foreach (var classDeclarationSyntax in userGeneratedClassesProvider)
+		{
+			var declarationSymbol = compilation.GetSymbol<INamedTypeSymbol>(classDeclarationSyntax);
+			if (declarationSymbol is null)
+			{
+				var diag = Diagnostic.Create(TextColorToDiagnostic.InvalidClassDeclarationSyntax, Location.None, classDeclarationSyntax.Identifier.Text);
+				context.ReportDiagnostic(diag);
+				continue;
+			}
+
+			if (declarationSymbol.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+				&& declarationSymbol.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default))
+			{
+				var nameSpace = declarationSymbol.ContainingNamespace.ToDisplayString();
+				if (nameSpace == "<global namespace>")
+				{
+					var diag = Diagnostic.Create(TextColorToDiagnostic.GlobalNamespace, Location.None, declarationSymbol.Name);
+					context.ReportDiagnostic(diag);
```

Comment:
I'm not much convinced by reporting no locations for diagnostics.

Reply:
I'm ok with not having a location here, since this isn't a code analyzer and I'm not scanning for a region to do a code fix, for example. I'm just telling the dev that a specific class should be inside a namespace. So the location here doesn't matter.

Reply:
@pictos Diagnostic location is not for codefixes only. It helps the developer double click the error to be navigated to the class he should fix.

Reply:
A location is also useful to suppress the diagnostic via `#pragma` directives.